cmp/cmpopts/equate.go

```diff
@@ -87,3 +88,25 @@ func areNaNsF64s(x, y float64) bool {
 func areNaNsF32s(x, y float32) bool {
 	return areNaNsF64s(float64(x), float64(y))
 }
+
+// EquateApproxTime returns a Comparer options that
+// determine two time.Time values to be equal if they
+// are within the given time interval of one another.
+//
+// The zero time is treated specially: it is only considered
+// equal to another zero time value.
```

Comment:
Rather than using a closure, I'd create a new `timeApproximator` type similar to the current `approximator` type. In the rare event that `cmp` panics due to conflicting options, the use of named functions and methods provides for nicer panic messages since the name will be used, rather than just some generic anonymous function name.

Reply:
I'm not sure I want to compare only non-zero times. For example it's not uncommon for a zero time value to be used as a sentinel value. I think it would be odd if a zero value were to cause the comparison to be skipped.

Reply:
> Rather than using a closure, I'd create a new timeApproximator type similar to the current approximator type. 

done

Reply:
>  I think it would be odd if a zero value were to cause the comparison to be skipped.

It won't be completely skipped. It will skip the application of *this* custom comparer. In that situation, `cmp` will fall back on the default behavior of `time.Time`, which is to use the `Equal` method (or another custom comparer if present).

The default `Equal` method does what you want:
```go
zero := time.Time{}
one := zero.Add(1)
fmt.Println(zero.Equal(zero)) // true
fmt.Println(zero.Equal(one))  // false

```

Reply:
Ah, that's cool. I just updated to exclude zero values; I didn't need to change the tests at all and the comparer logic is now a bit simpler.